From a755945a88a7ee35491fdb82f40f3d0f250d4aea Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Wed, 2 Dec 2020 11:48:59 +0700
Subject: [PATCH 204/358] arm64: dts: renesas: r8a774a1-hihope-rzg2m-rev2: add
 reserved memory regions

This commit adds reserved memory regions for HiHope RZ/G2M Rev.2 Board:
+ Global CMA:
	- 512 MiB : 0x0058000000 -> 0x0077ffffff
+ Lossy Decompression:
	- 48 MiB : 0x0054000000 -> 0x0056ffffff
+  CMA area for MMP:
	- 256 MiB : 0x0078000000 -> 0x0087ffffff

Also enable support mmngr and mmngrbuf drivers.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 .../dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts     | 36 ++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts
index 77a8f29..f80f23e 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts
@@ -23,6 +23,42 @@
 		device_type = "memory";
 		reg = <0x6 0x00000000 0x0 0x80000000>;
 	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/* device specific region for Lossy Decompression */
+		lossy_decompress: linux,lossy_decompress@54000000 {
+			no-map;
+			reg = <0x00000000 0x54000000 0x0 0x03000000>;
+		};
+
+		/* global autoconfigured region for contiguous allocations */
+		global_cma: linux,cma@58000000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x00000000 0x58000000 0x0 0x20000000>;
+			linux,cma-default;
+		};
+
+		/* device specific region for contiguous allocations */
+		mmp_reserved: linux,multimedia@70000000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x00000000 0x78000000 0x0 0x10000000>;
+		};
+	};
+
+	mmngr {
+		compatible = "renesas,mmngr";
+		memory-region = <&mmp_reserved>, <&lossy_decompress>;
+	};
+
+	mmngrbuf {
+		compatible = "renesas,mmngrbuf";
+	};
 };
 
 &du {
-- 
2.7.4

